### PR TITLE
Directed nocturnal

### PR DESCRIPTION
**Pull request by gzagatti**
### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [ ] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

This PR refactors `NocturnalCalls` to `NocturnalEvents` using `EventsTablesUnion` instead of `subscriber_locations` and allows subseting for direction.

### CODE REVIEW

File: flowmachine/flowmachine/features/subscriber/interevent_period.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import warnings
+from typing import List
+
+from ...utils.utils import verify_columns_exist_in_all_tables
+from ..utilities import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/interevent_period.py]: This seems to be from a different PR?

Comment (gzagatti) [reply flowmachine/flowmachine/features/subscriber/interevent_period.py]: I was building on top of previous branches, so it'd be better to approve them in order.

File: flowmachine/flowmachine/features/subscriber/nocturnal_events.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+"""
+from .metaclasses import SubscriberFeature
+from ..utilities import EventsTablesUnion
+
+
+class NocturnalEvents(SubscriberFeature):
+    """
+    Represents the percentage of events that a subscriber make/receives which
+    are at night. The definition of night is configurable.
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/nocturnal_events.py]:
```suggestion
    began at night. The definition of night is configurable.
```
